astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+            else:
+                col.meta.unit = "---"
+
+            # Add col limit values to col description
+            lim_vals = ""
+            if (col.min and col.max and
+                not any(x in col.name for x in ['RA', 'DE', 'LON', 'LAT'])):
+                # No col limit values for coordinate columns.
+                if col.fortran_format[0] == 'I':
+                    if abs(col.min) < MAX_COL_INTLIMIT and abs(col.max) < MAX_COL_INTLIMIT:
+                        if col.min == col.max:
+                            lim_vals = "[{0}]".format(col.min)
+                        else:
+                            lim_vals = "[{0}/{1}]".format(col.min, col.max)
+                elif col.fortran_format[0] in ('E', 'F'):
+                    lim_vals = "[{0}/{1}]".format(math.floor(col.min * 100) / 100.,
+                                                  math.ceil(col.max * 100) / 100.)
+
+            if lim_vals != '' or nullflag != '':
+                description = "{0}{1} {2}".format(lim_vals, nullflag, description)
+
+            # Find the maximum label and description column widths.
+            if len(col.name) > max_label_width:
+                max_label_width = len(col.name)
+            if len(description) > max_descrip_size:
+                max_descrip_size = len(description)
+
+            # Add a row for the Sign of Declination in the bbb table
+            if col.name == 'DEd' and col[0] < 0.0:
+                bbb.add_row([singlebfmt.format(startb),
+                             "A1", "---", "DE-",
+                             "Sign of Declination"])
+                startb += 1
+
+            # Add Byte-By-Byte row to bbb table
+            bbb.add_row([singlebfmt.format(startb) if startb == endb
+                            else fmtb.format(startb, endb),
+                         "" if col.fortran_format is None else col.fortran_format,
+                         col.meta.unit,
+                         "" if col.name is None else col.name,
+                         description])
+            startb = endb + 2
+
+        # Properly format bbb columns
+        bbblines = StringIO()
+        bbb.write(bbblines, format='ascii.fixed_width_no_header',
+                  delimiter=' ', bookend=False, delimiter_pad=None,
+                  formats={'Format': '<6s',
+                           'Units': '<6s',
+                           'Label': '<'+str(max_label_width)+'s',
+                           'Explanations': ''+str(max_descrip_size)+'s'})
+
+        # Get formatted bbb lines
+        bbblines = bbblines.getvalue().splitlines()
+
+        # ``nsplit`` is the number of whitespaces to prefix to long description
+        # lines in order to wrap them. It is the sum of the widths of the
+        # previous 4 columns plus the number of single spacing between them.
+        # The hyphen in the Bytes column is also counted.
+        nsplit = byte_count_width*2 + 1 + 12 + max_label_width + 4
```

Comment:
This logic is not quite robust for giving a strictly-formatted header, though I guess it is probably legal from the examples I've seen.
```
t = Table()
t['a'] = ['a'*80, 'b'*80]
t['b'] = ['a'*80, 'b'*80]
t['a'].info.description = ' '.join(['casdfasdf'] * 20)
ascii.write(t, format='cds')

================================================================================
Byte-by-byte Description of file: table.dat
--------------------------------------------------------------------------------
 Bytes Format Units  Label     Explanations
--------------------------------------------------------------------------------
  1- 80  A80    ---    a       casdfasdf casdfasdf casdfasdf casdfasdf casdfasdf
                              casdfasdf casdfasdf casdfasdf casdfasdf casdfasdf
                              casdfasdf casdfasdf casdfasdf casdfasdf casdfasdf
                              casdfasdf casdfasdf casdfasdf casdfasdf casdfasdf
 82-161  A80    ---    b       Description of b
--------------------------------------------------------------------------------
```

Reply:
Haan, the alignment of Byte-By-Byte mini-table header is not properly set. It is good to have that header line aligned though. So, I have done modifications to align it in one of the later PRs.